### PR TITLE
HubSpot CRM (Independent Publisher) Minor corrections to existing operations, ContactsSearch operation

### PR DESCRIPTION
**Pull request by ryanedshannon**
Changes:
* Added ContactsSearch operation to allow searching for a set of contacts using a set of filters
* Added optional properties field to ContactsRead to define what fields to return
* Fixed issue with requesting specific properties in ContactsList operation
* Fixed typo on properties in CompaniesRead operation

ContactsSearch1
ContactsSearch2
ContactsSearch3

ContactsRead

ContactsList

CompaniesRead

### CODE REVIEW

**Comment by ryanedshannon** [comment]
@microsoft-github-policy-service agree company="Ricoh DTG"

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 1]
```diff
@@ -2803,6 +2813,126 @@
         ]
       }
     },
+    "/crm/v3/objects/contacts/search": {
+      "post": {
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:2819]: @ryanedshannon it is not allowed to have a default response only. please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 2]
```diff
@@ -2803,6 +2813,126 @@
         ]
       }
     },
+    "/crm/v3/objects/contacts/search": {
+      "post": {
+        "responses": {
+          "default": {
+            "description": "The default response.",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:2821]: @ryanedshannon empty schema is not allowed. please define a proper schema.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]
@ryanedshannon Thank you so much for your contribution to this connector.
I would like to bring your attention to an existing [PR ](https://github.com/microsoft/PowerPlatformConnectors/pull/2533)for this connector which is already being deployed. Unfortunately, we can't have two deployments for the same connector at the same time. Therefore, we will put this update "OnHold" until the current deployment is complete.
Please let me know if you have any questions.
Thank you so much again.

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json (hunk 1 above)
Comment (ryanedshannon) [reply independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:2859]: Thank you for the review and feedback. I've updated with a set of responses. Let me know if this is sufficient.

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json (hunk 2 above)
Comment (ryanedshannon) [reply independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json]: I defined the "ErrorModel" schema with the standard HubSpot structure and used it as a reference in the search operation. Let me know if that looks good.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @ryanedshannon ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 3]
```diff
@@ -32,502 +104,140 @@
...
+        ]
+      }
+    },
+    "/crm/v3/objects/companies/{companyId}": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK",
             "schema": {
-              "type": "object",
```
Comment (ryanedshannon) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:84]: Moved to definitions

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 4]
```diff
@@ -580,490 +319,118 @@
...
-            "required": true,
-            "type": "string",
-            "description": "The companyId of the company.",
-            "x-ms-visibility": "important",
-            "x-ms-summary": "companyId",
-            "x-ms-url-encoding": "single"
+            "$ref": "#/parameters/CompanyID"
           },
           {
-            "name": "properties",
```
Comment (ryanedshannon) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:616]: Moved to definitions

**Comment by ryanedshannon** [comment]
Hello @Amjed-Ayoub,

I just noticed I accidentally added security definitions for API authorization method (I couldn't get oauth to work with HubSpot).
Should I remove these from this request? Also, should I still be adding to this request or holding off until it is deployed?

Thanks.

**Comment by Amjed-Ayoub** [comment]
@ryanedshannon yes please fix and get this PR ready for the next deployment. I will move forward with it once the current one is complete.

**Comment by Amjed-Ayoub** [comment]
Hello @ryanedshannon ,
I hope you are doing well.
Congratulations, the current deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
@ryanedshannon we will start with the new deployment today :-).

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 5]
```diff
@@ -32,588 +91,895 @@
...
+          }
+        ]
+      }
+    },
+    "/crm/v3/objects/deals": {
+      "get": {
         "responses": {
           "default": {
             "description": "The default response.",
             "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json]: @ryanedshannon schema is not allowed within the default response. please remove it.  Please fix all default responses in the swagger file.

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json [hunk 6]
```diff
@@ -1331,615 +1376,313 @@
...
+              }
+            }
+          }
+        ]
+      }
+    },
+    "/crm/v3/objects/products": {
+      "get": {
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:1135]: @ryanedshannon it is not allowed to have a default response only. Please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses). please fix all responses in the swagger file.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json (hunk 5 above)
Comment (ryanedshannon) [reply independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json]: @Amjed-Ayoub I believe that's all responses accounted for, let me know if there are any other issues. Thank you for the help.

File: independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json (hunk 6 above)
Comment (vmanoharas) [reply independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:1135]: Hello @ryanedshannon, can you please fix the comment added by Amjed above.

Comment (ryanedshannon) [reply independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:1135]: Hello @vmanoharas, I think this should already be resolved, I don't see any other responses left with this issue.

**Comment by vmanoharas** [comment]
Hello @ryanedshannon 
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by vmanoharas — APPROVED** [review]

**Pull request merged**